runtime/wasmer_test.go

```diff
@@ -662,12 +747,46 @@ func TestExt_twox_128(t *testing.T) {
 	}
 
 	//check result against expected value
-	t.Logf("Ext_twox_128 data: %s, result: %s", data, hex.EncodeToString(mem[out:out+16]))
 	if "b27dfd7f223f177f2a13647b533599af" != hex.EncodeToString(mem[out:out+16]) {
 		t.Error("hash saved in memory does not equal calculated hash")
 	}
 }
 
+// test that ext_keccak_256 returns the correct hash
+func TestExt_keccak_256(t *testing.T) {
+	runtime, err := newTestRuntime()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	mem := runtime.vm.Memory.Data()
+
+	data := []byte(nil)
+	pos := 170
+	out := pos + len(data)
+	copy(mem[pos:pos+len(data)], data)
+
+	// call wasm function
+	testFunc, ok := runtime.vm.Exports["test_ext_keccak_256"]
+	if !ok {
+		t.Fatal("could not find exported function")
+	}
+
+	_, err = testFunc(pos, len(data), out)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expected, err := common.HexToHash("0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470")
```

Comment:
magical value

Reply:
this is from here: https://github.com/debris/tiny-keccak/blob/master/tests/keccak.rs#L4 I'll document it

Reply:
added comment to document